### PR TITLE
`Assessment`: Only notify students at assessment due date if they actually received an assessment

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.

### Motivation, Context and Description

Fix a bug where all students received a notification at the assessment due date, even if their assessment was not finished.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with manual assessment

1. Log in to Artemis
2. Participate with both students in the exercise
3. Assess one before the **assessment** due date
4. Check that only the student with an assessment got the notification at the assessment due date
5. Assess the other submission
6. Check that the other student now has a notification as well

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java [hunk 1]
```diff
@@ -241,17 +246,21 @@ void testNotifyUsersAboutAssessedExerciseSubmission() {
...
+        manualResult.setAssessmentType(AssessmentType.MANUAL);
+        resultRepository.save(manualResult);
         database.createAndSaveParticipationForExercise(testExercise, studentWithParticipationButWithoutSubmission.getLogin());
 
         testExercise = exerciseRepository.findAllExercisesByCourseId(testCourse.getId()).iterator().next();
 
         singleUserNotificationService.notifyUsersAboutAssessedExerciseSubmission(testExercise);
 
-        assertThat(notificationRepository.findAll()).as("Only one notification should have been created (for the user with a valid participation, submission, and result)")
+        assertThat(notificationRepository.findAll()).as("Only one notification should have been created (for the user with a valid participation, submission, and manual result)")
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/notification/SingleUserNotificationServiceTest.java:263]: Can you maybe also ensure that the notification actually got sent to only student3, and not student1?

**Review by Strohgelaender — APPROVED** [review]
looks good

**Review by Strohgelaender — APPROVED** [review]

**Review by timo07k — COMMENTED** [review]
Sorry I misunderstood the task

**Review by timo07k — APPROVED** [review]
Everything works as expected. Tested on TS1.

**Review by PhilipWerz — APPROVED** [review]
Everything works as expected. Tested on TS1.

Remark:
I didn't get a pop up notification for the 4. testing step but notification is shown under notification bell. Got a pop up (slide) Notfication from the 6. step

**Review by Vinicius-Agreste — APPROVED** [review]
Tested on ts2. Working as expected.

**Review by TheZoker — APPROVED** [review]
Code LGTM 👍

**Pull request merged**